libs/Microsoft.MixedReality.WebRTC.Native/test/external_video_track_source_tests.cpp

```diff
@@ -112,30 +116,134 @@ void ValidateQuadTestFrame(const void* data,
+          mrsRemoteVideoTrackHandle track_handle,
+          mrsTransceiverInteropHandle /*interop_handle*/,
+          mrsTransceiverHandle transceiver_handle) {
+        track_handle2 = track_handle;
+        transceiver_handle2 = transceiver_handle;
+        track_added2_ev.Set();
+      };
+  mrsPeerConnectionRegisterVideoTrackAddedCallback(pair.pc2(),
+                                                   CB(track_added2_cb));
 
-  ExternalVideoTrackSourceHandle source_handle = nullptr;
+  // Create the external source for the local video track of the local peer (#1)
+  mrsExternalVideoTrackSourceHandle source_handle1 = nullptr;
   ASSERT_EQ(mrsResult::kSuccess,
             mrsExternalVideoTrackSourceCreateFromArgb32Callback(
-                &GenerateQuadTestFrame, nullptr, &source_handle));
-  ASSERT_NE(nullptr, source_handle);
-  mrsExternalVideoTrackSourceFinishCreation(source_handle);
+                &GenerateQuadTestFrame, nullptr, &source_handle1));
+  ASSERT_NE(nullptr, source_handle1);
+  mrsExternalVideoTrackSourceFinishCreation(source_handle1);
 
-  LocalVideoTrackHandle track_handle = nullptr;
-  LocalVideoTrackFromExternalSourceInitConfig source_config{};
-  ASSERT_EQ(mrsResult::kSuccess,
-            mrsPeerConnectionAddLocalVideoTrackFromExternalSource(
-                pair.pc1(), "gen_track", source_handle, &source_config,
-                &track_handle));
-  ASSERT_NE(nullptr, track_handle);
-  ASSERT_NE(mrsBool::kFalse, mrsLocalVideoTrackIsEnabled(track_handle));
+  // Create the local track itself for #1
+  mrsLocalVideoTrackHandle track_handle1{};
+  {
+    mrsLocalVideoTrackFromExternalSourceInitConfig source_config{};
+    ASSERT_EQ(mrsResult::kSuccess,
+              mrsLocalVideoTrackCreateFromExternalSource(
+                  source_handle1, &source_config, "gen_track", &track_handle1));
+    ASSERT_NE(nullptr, track_handle1);
+    ASSERT_NE(mrsBool::kFalse, mrsLocalVideoTrackIsEnabled(track_handle1));
+  }
+
+  // Create the video transceiver #1
+  mrsTransceiverHandle transceiver_handle1{};
+  {
+    mrsTransceiverInitConfig config{};
+    config.name = "transceiver_1";
+    ASSERT_EQ(mrsResult::kSuccess,
+              mrsPeerConnectionAddVideoTransceiver(pair.pc1(), &config,
+                                                   &transceiver_handle1));
+    ASSERT_NE(nullptr, transceiver_handle1);
+  }
+
+  // Check video transceiver #1 consistency
+  {
+    // Local track is NULL
+    mrsLocalVideoTrackHandle track_handle_local{};
+    ASSERT_EQ(Result::kSuccess, mrsTransceiverGetLocalVideoTrack(
+                                    transceiver_handle1, &track_handle_local));
+    ASSERT_EQ(nullptr, track_handle_local);
+
+    // Remote track is NULL
+    mrsRemoteVideoTrackHandle track_handle_remote{};
+    ASSERT_EQ(Result::kSuccess, mrsTransceiverGetRemoteVideoTrack(
+                                    transceiver_handle1, &track_handle_remote));
+    ASSERT_EQ(nullptr, track_handle_remote);
+  }
```

Comment:
nit: IMHO we can just have one test that checks if a new transceiver is consistent, and then we can trust that all new transceivers are consistent - we don't need to repeat this block in every test that deals with transceivers.

Reply:
Fixed.